crates/core/src/operations/update.rs

```diff
@@ -597,24 +1050,144 @@ pub(crate) async fn request_update(
             let new_state = Some(UpdateState::AwaitingResponse {
                 key,
                 upstream: None,
+                retry_count: 0,
             });
             let msg = UpdateMsg::RequestUpdate {
                 id,
                 key,
-                related_contracts,
-                target,
-                value,
+                related_contracts: related_contracts.clone(),
+                target: target.clone(),
+                value: value.clone(),
             };
 
             let op = UpdateOp {
                 state: new_state,
                 id,
-                stats: update_op.stats,
+                stats: update_op.stats.clone(),
             };
 
-            op_manager
+            match op_manager
+                .notify_op_change(NetMessage::from(msg), OpEnum::Update(op))
+                .await
+            {
+                Ok(_) => {
+                    tracing::debug!("Successfully sent initial update request for contract {}", key);
+                }
+                Err(err) => {
+                    tracing::warn!(
+                        "Failed to send initial update request for contract {}: {}. Will retry.",
+                        key,
+                        err
+                    );
+                    
+                    let retry_state = Some(UpdateState::RetryingRequest {
+                        key,
+                        target,
+                        related_contracts,
+                        value,
+                        retry_count: 0,
+                    });
+                    
+                    let retry_op = UpdateOp {
+                        state: retry_state,
+                        id,
+                        stats: update_op.stats.clone(),
+                    };
+                    
+                    op_manager
+                        .notify_op_change(
+                            NetMessage::from(UpdateMsg::AwaitUpdate { id }),
+                            OpEnum::Update(retry_op),
+                        )
+                        .await?;
+                }
+            }
+        }
+        Some(UpdateState::RetryingRequest {
+            key,
+            target: retry_target,
+            related_contracts,
+            value,
+            retry_count,
+        }) => {
+            let delay_ms = std::cmp::min(
```

Comment:
instead of doing this, just reuse the utils we have in crates/core/src/util/mod.rs for backoff